kernel/init.c

```diff
@@ -197,7 +197,7 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 #endif
 	boot_banner();
 
-#ifdef CONFIG_CPLUSPLUS
+#if defined(CONFIG_CPLUSPLUS) && !defined(CONFIG_BOARD_NATIVE_POSIX)
```

Comment:
Thanks for looking into this.

Since this is applicable to all "POSIX arch" platforms and not just `native_posix` board, it should be revised as follows:
```suggestion
#if defined(CONFIG_CPLUSPLUS) && !defined(CONFIG_ARCH_POSIX)
```